src/go/k8s/apis/cluster.redpanda.com/v1alpha1/topic_types.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2021-2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package v1alpha1
+
+import (
+	"time"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TopicSpec defines the desired state of Topic
+type TopicSpec struct {
+	// Partitions is the number topic shards that is distributed across the nodes in a cluster.
+	// This cannot be decreased after topic creation.
+	// It can be increased after topic creation, but it is
+	// important to understand the consequences that has, especially for
+	// topics with semantic partitioning. When absent this will default to
+	// the Redpanda cluster configuration `default_topic_partitions`.
+	// https://docs.redpanda.com/docs/reference/cluster-properties/#default_topic_partitions
+	// https://docs.redpanda.com/docs/get-started/architecture/#partitions
+	Partitions *int `json:"partitions,omitempty"`
+	// ReplicationFactor is the number of replicas the topic should have. Must be odd value.
+	// When absent this will default to the Redpanda cluster configuration `default_topic_replications`.
+	// https://docs.redpanda.com/docs/reference/cluster-properties/#default_topic_replications
+	ReplicationFactor *int `json:"replicationFactor,omitempty"`
+	// OverwriteTopicName will change the topic name from the `metadata.name` to `OverwriteTopicName`
+	OverwriteTopicName *string `json:"overwriteTopicName,omitempty"`
+	// AdditionalConfig is free form map of any configuration option that topic can have.
+	// Examples:
+	// cleanup.policy=compact
+	// redpanda.remote.write=true
+	// redpanda.remote.read=true
+	// redpanda.remote.recovery=true
+	// redpanda.remote.delete=true
+	AdditionalConfig map[string]*string `json:"additionalConfig,omitempty"`
+
+	// KafkaAPISpec act as explicit Kafka API client configuration to Redpanda
+	KafkaAPISpec *KafkaAPISpec `json:"kafkaApiSpec,omitempty"`
+
+	// MetricsNamespace can be used to overwrite fully-qualified
+	// name of the Metric. That should be easier to identify if
+	// multiple operator runs inside the same Kubernetes cluster.
+	// By default, it is set to `redpanda-operator`.
+	MetricsNamespace *string `json:"metricsNamespace,omitempty"`
+
+	// SynchronizationInterval when the topic controller will schedule next reconciliation
+	// Default is 3 seconds
+	SynchronizationInterval *time.Duration `json:"interval,omitempty"`
```

Comment:
> Do we need this on a per-topic basis or can we just set it as an operator flag?

You can manage multiple Redpanda cluster instances using topic custom resource. In each reconciliation controller opens dedicated connection to one Redpanda cluster. If customer would like to treat Production environment differently than Development environment this option gives that ability. 

Another argument could be that operation team gives developers team a chance to set up synchronization interval. Each dev team could have different SLO, so with your proposal each developer team needs dedicated operator. That would put more maintenance on operation team.

That's why I proposed interval to be in the Topic definition.

> Why do we do this every 3 seconds?

Mainly, because default kubernetes controller runtime is 1 hour and we don't have a notification channel that could trigger our reconciliation loop (like Kubernetes watch mechanism can).

I'm open for suggestion to change the default. Maybe you would like to see default property being exposed as a flag.